bridge/src/lib.rs

```diff
@@ -3,11 +3,78 @@ extern crate rutie;
 extern crate lazy_static;
 
 mod connection;
+mod worker;
 
 use connection::Connection;
-use rutie::{Module, Object, Symbol, RString, Encoding, AnyObject, AnyException, Exception, VM, Thread};
+use once_cell::sync::OnceCell;
+use rutie::{
+    Module, Object, Symbol, RString, Encoding, AnyObject, AnyException, Exception, VM, Thread,
+    NilClass
+};
+use std::cell::RefCell;
+use std::sync::Arc;
+use std::sync::mpsc::{sync_channel, SyncSender, Receiver};
+use std::time::Duration;
+use temporal_sdk_core::{Logger, TelemetryOptionsBuilder};
+use tokio::runtime::{Builder, Runtime};
+use worker::{Response, Worker, WorkerResult};
+
+const RUNTIME_THREAD_COUNT: u8 = 2;
+
+fn runtime() -> &'static Arc<Runtime> {
+    static RUNTIME: OnceCell<Arc<Runtime>> = OnceCell::new();
+    RUNTIME.get_or_init(|| {
+        Arc::new(
+            Builder::new_multi_thread()
+                .worker_threads(RUNTIME_THREAD_COUNT.into())
+                .enable_all()
+                .thread_name("core")
+                .build()
+                .expect("Unable to start a runtime")
+        )
+    })
+}
+
+fn raise_bridge_exception(message: &str) {
+    VM::raise_ex(AnyException::new("Temporal::Bridge::Error", Some(message)));
+}
+
+fn run_callback_loop(receiver: &mut Receiver<Response>) {
+    let timeout = Duration::from_secs(5);
+
+    while let Ok(msg) = Thread::call_without_gvl(|| { receiver.recv_timeout(timeout) }, Some(|| {})) {
+        match msg {
+            Response::ActivityTask { bytes, callback } => callback(Ok(bytes)),
+            Response::Error { error, callback } => callback(Err(error)),
+            _ => panic!("Unsupported response type"),
+        }
+    }
+}
+
+fn start_callback_loop() -> &'static SyncSender<Response> {
+    static CALLBACK_SENDER: OnceCell<SyncSender<Response>> = OnceCell::new();
+    CALLBACK_SENDER.get_or_init(|| {
+        let (tx, rx): (SyncSender<Response>, Receiver<Response>) = sync_channel(1);
```

Comment:
Clone. From https://doc.rust-lang.org/std/sync/mpsc/struct.Sender.html:

> This half can only be owned by one thread, but it can be cloned to send to other threads.

Reply:
Just clone the send side. This channel doesn't need to be static.